cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -691,6 +707,19 @@ public Collection<ApplicationSpecification> getAllAppVersions(ApplicationId id)
     });
   }
 
+  @Override
+  public ApplicationMeta getLatestAppVersion(ApplicationId id) {
+    // TODO: Change this to sort by creation time instead of version.
```

Comment:
Note: this is a temporary fix - we will not scan the table - but instead use the index on 'creation time' when the field is added to AppMetadataStore app-spec table